packages/base/src/3dview/helpers.ts

```diff
@@ -94,31 +94,32 @@ export function computeExplodedState(options: {
 }) {
   const { mesh, boundingGroup, factor } = options;
   const center = new THREE.Vector3();
+  const parent = mesh.parent as THREE.Object3D;
   boundingGroup.getCenter(center);
 
-  const oldGeometryCenter = new THREE.Vector3();
-  mesh.geometry.boundingBox?.getCenter(oldGeometryCenter);
+  const meshBoundingBox = new THREE.Box3().setFromObject(mesh);
+  let oldGeometryCenter = new THREE.Vector3();
+  meshBoundingBox.getCenter(oldGeometryCenter);
+  oldGeometryCenter = oldGeometryCenter.add(parent.position);
+  console.log('oldGeometryCenter', oldGeometryCenter);
+  
 
-  const centerToMesh = new THREE.Vector3(
-    oldGeometryCenter.x - center.x,
-    oldGeometryCenter.y - center.y,
-    oldGeometryCenter.z - center.z
-  );
-  const distance = centerToMesh.length() * factor;
+  const centerToMesh = new THREE.Vector3()
+    .subVectors(oldGeometryCenter, center)
+    .normalize();
 
-  centerToMesh.normalize();
+  const distance = centerToMesh.multiplyScalar(factor);
```

Comment:
That looks incorrect.

It used to be `distance = centerToMesh.length() * factor;` but it is now `distance = centerToMesh.normalize() * factor;`